extensions/sql-database-projects/src/models/project.ts

```diff
@@ -170,6 +170,13 @@ export class Project implements ISqlProject {
 						} else {
 							// only add file if it wasn't already added
 							if (!this._files.find(f => f.relativePath === relativePath)) {
+								// SSDT encodes @ to %40, so we need to decode that so that it shows correctly in the tree
+								// I didn't find any other characters that get encoded, so this check is specifically for %40 to have lower risk of accidentally breaking something
+								// but we can decode all relative paths in the future if we find other characters that get encoded
+								if (relativePath.includes('%40')) {
```

Comment:
But this is just SSDT doing weird behavior right? What about if the user has a path with %40 in it themselves? I'm not sure we should be assuming that all project files have this behavior. 

Unfortunately it's going to be pretty much impossible to actually know for sure whether something is %40 encoded or not - since we have no way of knowing whether the %40 was actually intended by the user or an artifact of something else encoding the path. 

So I guess I'm not sure what the actual best fix here is. But it seems like SSDT doing this is more of the root issue here and finding out why that's being done first would be good.

Reply:
That's a good point. SSDT doesn't allow adding a new item with %40 in the name since % isn't a valid file character. If we were just on windows, then it would be ok to assume %40 was encoded since it's an invalid file character, but we couldn't be sure for mac or linux :/

Looks like we don't currently have any name validation in ADS, so I opened https://github.com/microsoft/azuredatastudio/issues/17654. Since we make sure our projects work in both ADS and VS, maybe the file name validation make sure file names are valid on windows, even if ADS is being used on mac. It's definitely a use case by some customers that the same sql project is used on mac and windows, so we'd want to make sure the project works and file names are valid on both platforms.

Reply:
The bug also reports (https://github.com/microsoft/azuredatastudio/issues/17020#issuecomment-918868348) that this was occurring in a project that was reverse-engineered four years ago, and that newly-added procedures don't have this issue.  They were also able to easily work around it with some find-replace magic in the sqlproj file.

I think we shouldn't bother addressing this because it's no longer a bug in SSDT, and instead close the bug with repair steps (find-replace).

Reply:
% is a valid file character on Linux and Windows at least, where are you seeing that it's not?

Reply:
It's SSDT (or VS) that disallows it, not the OS: 
![image](https://user-images.githubusercontent.com/1609827/141382306-0273018b-e436-4605-b957-bfa20aca5310.png)